dev_guide/builds.adoc

```diff
@@ -511,6 +514,65 @@ customStrategy:
 ----
 ====
 
+[[pipeline-strategy-options]]
+
+== Pipeline Strategy Options
+
+The following options are specific to the
+xref:../architecture/core_concepts/builds_and_image_streams.adoc#pipeline-build[Pipeline
+build strategy].
+
+[[jenkinsfile]]
+
+=== Providing the Jenkinsfile
+
+Pipeline builds require a Jenkinsfile which defines the steps that make
+up the pipeline that will be executed.  The Jenkinsfile can be provided
+in one of two ways.
+
+The first option is to include the Jenkinsfile definition directly in the build configuration,
+define the jenkinsfile field:
+
+====
+[source,yaml]
+----
+kind: "BuildConfig"
+apiVersion: "v1"
+metadata:
+  name: "sample-pipeline"
+spec:
+  strategy:
+    type: "JenkinsPipeline"
+    jenkinsPipelineStrategy:
+      jenkinsfile: "node('agent') {\nstage 'build'\nopenshiftBuild(buildConfig: 'ruby-sample-build', showBuildLogs: 'true')\nstage 'deploy'\nopenshiftDeploy(deploymentConfig: 'frontend')\n}"
+----
+====
+
+The second option is to include a reference to a git repository which contains the Jenkinsfile:
+
+
+====
+[source,yaml]
+----
+kind: "BuildConfig"
+apiVersion: "v1"
+metadata:
+  name: "sample-pipeline"
+spec:
+  source:
+    type: "Git"
+    git:
+      uri: "https://github.com/openshift/ruby-hello-world"
+  strategy:
+    type: "JenkinsPipeline"
+    jenkinsPipelineStrategy:
+      jenkinsfilePath: some/repo/dir <1>
+----
+
+<1> An optional path within the repository that points to the Jenkinsfile.  This path is relative to the source `*contextDir*`, 
+if any.  If left empty, the file must reside in the root of the repository or `*contextDir*`.
```

Comment:
sorry.  misinterpretted your answer.  so it's a full path to the exact file to be used.